qiskit/optimization/converters/inequality_to_equality_converter.py

```diff
@@ -152,6 +153,10 @@ def encode(self, op: OptimizationProblem, name: Optional[str] = None,
             else:
                 raise QiskitOptimizationError('Type of sense in ' + variable + 'is not supported')
 
+        # TODO: add quadratic constraints
+        if self._src.quadratic_constraints.get_num() > 0:
+            warnings.warn("Quadratic constraints are currently not suppourted yet.")
```

Comment:
Same, here maybe this should raise an error instead of a warning if a user tries to do a conversion we do not support yet.

Reply:
hmm, you're right. I think raising an error is more natural. I will fix it.